src/theme-context-provider.tsx

```diff
@@ -171,7 +161,25 @@ const ThemeContextProvider: React.FC<Props> = ({theme, children, as, withoutStyl
         [colors]
     );
 
+    // TODO: create CSS vars for size and lineHeight (https://jira.tid.es/browse/WEB-1929)
+    const textPresetsVars = React.useMemo(() => {
+        // Get an object mapping textPresets tokens to objects containing the token's weight
+        // For example, {title1: {weight: '700'}}
+        const textTokens = Object.keys(contextTheme.textPresets) as Array<keyof TextPresetsConfig>;
+        const tokenValues = textTokens.map((token) => {
+            // Map light/regular/medium/bold to valid css fontWeight values
+            return {[token]: {weight: String(mapToWeight[contextTheme.textPresets[token].weight])}};
+        });
```

Comment:
You could use `Object.entries(...).map` and condense these two statements

Reply:
done